From d7ac282702fb26f0537516b8fc6c6f1d5f5ec5f6 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 22 Apr 2022 20:18:14 +0200
Subject: [PATCH 16/26] arm64: dts: imx8mp: Enable HDMI on MX8MP DHCOM PDK2 and
 PDK3

Enable HDMI output on MX8MP DHCOM PDK2. The I2C5 is used in regular I2C
mode instead of HDMI DDC mode to permit connection of other I2C devices
on that bus.

Enable HDMI output on MX8MP DHCOM PDK3. The I2C mux port 1 is used
instead of HDMI DDC I2C5 to permit connection of other I2C devices
on that bus.

Upstream-Status: Pending [Depends on MX8MP HDMI support]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts | 11 +++++++++++
 arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts | 11 +++++++++++
 2 files changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
index e9fb5f7f39b50..f4e39bc0f1657 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk2.dts
@@ -184,6 +184,17 @@ &flexcan1 {
 	status = "okay";
 };
 
+&hdmi_tx {
+	ddc-i2c-bus = <&i2c5>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi>;
+	status = "okay";
+};
+
+&hdmi_tx_phy {
+	status = "okay";
+};
+
 &pcie_phy {
 	clock-names = "ref";
 	clocks = <&clk IMX8MP_SYS_PLL2_100M>;
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
index 31d85d5871c91..8fbb86499edac 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-dhcom-pdk3.dts
@@ -283,6 +283,17 @@ &flexcan1 {
 	status = "okay";
 };
 
+&hdmi_tx {
+	ddc-i2c-bus = <&i2cmuxed1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi>;
+	status = "okay";
+};
+
+&hdmi_tx_phy {
+	status = "okay";
+};
+
 &pcie_phy {
 	clocks = <&pcieclk 1>;
 	clock-names = "ref";
-- 
2.43.0

